sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -247,24 +239,38 @@ async static Task CreateStoredAccessPolicyAsync(string containerName)
 
-[//]: # (Blob Metadata, properties, and attributes...)
+The modern SDK has taken a client-based approach. There are no objects designed to be representations of storage resources, but instead clients that act as your mechanism to interact with your storage resources in the cloud. Clients hold no state of your resources.
 
-### Client constructors
+The hierarchical structure of Azure Blob Storage can be understood by the following diagram:
+![Blob Storage Hierarchy](https://docs.microsoft.com/en-us/azure/storage/blobs/media/storage-blobs-introduction/blob1.png)
+
+In the interest of simplifying the API surface, v12 uses three top level clients to match this structure that can be used to interact with a majority of your resources: `BlobServiceClient`, `BlobContainerClient`, and `BlobClient`. Note that blob-type-specific operations can still be accessed by their specific clients, as in v11.
+
+`BlobClient` is NOT a replacement of v11's `CloudBlob`, which served as a base class for the three blob type classes. v12 contains `BlobBaseClient` as a replacement base class. `BlobClient` is a SDK-only concept that trades off advanced functionality like partial blob updates for an easier to understand abstraction of blobs, wher you do not need to worry about blob types or their implementation mechanisms. They are backed up by block blobs in Storage, and you can switch your code over to using `BlockBlobClient`s with no extra steps involving your already-stored data.
+
+Note the absence of a v12 equivalent for v11's `CloudBlobDirectory`. Directories were an SDK-only concept that did not exist Azure Blob Storage, and which were not brought forwards into the modern Storage SDK. As shown by the diagram above, containers only contain a flat list of blobs, but those blobs can be named and listed in ways that imply a folder-like structure. See our [Listing Blobs in a Container](#listing-blobs-in-a-container) migration samples later in this guide for more information.
+
+For those whose workloads revolve around manipulating directories and heavily relied on the leagacy SDKs abstraction of this structure, consider the [pros and cons of enabling hierarchical namespace](https://docs.microsoft.com/azure/storage/blobs/data-lake-storage-namespace) on your storage account, which would allow switching to the [Data Lake gen 2 SDK](https://docs.microsoft.com/dotnet/api/overview/azure/storage.files.datalake-readme), whose migration is not covered in this document.
+
+The following table lists v11 classes and their v12 equivalents for quick reference.
 
 | v11 | v12 |
 |-------|--------|
-| `CloudStorageAccount` | `BlobServiceClient` |
+| `CloudBlobClient` | `BlobServiceClient` |
 | `CloudBlobContainer`  | `BlobContainerClient` |
-| `CloudBlobDirectory` | Not supported |
+| `CloudBlobDirectory` | No equivalent |
+| No equivalent | `BlobClient`
 | `CloudBlob` | `BlobBaseClient` |
```

Comment:
I'd rather rephrase this section in such a way that user is encouraged to try `BlobClient` first. Suggesting that there's no equivalent and saying that BlobClient is not a replacement for CloudBlob above might discourage from trying it first.

Maybe we should say that if they don't have any peculiar requirements then they should be trying BlobClient first?

Reply:
I added two subsections to this section. One of them is "Migrating from CloudBlockBlob", who's opening paragraph is "We recommend `BlobClient` as a starting place when migrating code that used v11's `CloudBlockBlob`."

Reply:
looks good.